jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -68,13 +68,22 @@ YUI.add('juju-env-bakery', function(Y) {
         } else {
           this.visitMethod = this._defaultVisitMethod;
         }
-        this.macaroonName = 'Macaroons-' + cfg.serviceName;
+        if (cfg.existingCookie) {
+          this.macaroonName = cfg.existingCookie;
+        } else {
+          this.macaroonName = 'Macaroons-' + cfg.serviceName;
+        }
         this.staticMacaroonPath = cfg.staticMacaroonPath;
         this.setCookiePath = cfg.setCookiePath;
         this.nonceLen = 24;
+        this.nonCookie = !!cfg.nonCookie;
```

Comment:
personally I'm a fan of using 'truthy' statements so that would be `useLocalStorage` instead of `dontUseCookie`. It's a little easier to understand, especially for non-English speakers.